src/model/interactions.rs

```diff
@@ -0,0 +1,108 @@
+//! Interactions information-related models
+
+use super::prelude::*;
+use crate::internal::prelude::*;
+
+/// Information about an Interaction
+///
+/// An interaction is the base "thing" that is sent when a user invokes a command, and is the same
+/// for Slash Commands and other future interaction types.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[non_exhaustive]
+pub struct Interaction {
+    pub id: InteractionId,
+    #[serde(rename = "type")]
+    pub kind: InteractionType,
+    pub data: Option<ApplicationCommandInteractionData>,
+    pub guild_id: GuildId,
+    pub channel_id: ChannelId,
+    pub member: Member,
+    pub token: String,
+    pub version: i32
+}
+
+/// The type of an Interaction
+#[derive(Copy, Clone, Debug, Deserialize, Hash, Eq, PartialEq, PartialOrd, Ord, Serialize)]
+#[non_exhaustive]
+#[repr(u8)]
+pub enum InteractionType {
+    Ping = 1,
+    ApplicationCommand = 2
```

Comment:
```suggestion
    ApplicationCommand = 2,
```
Missing trailing comma.